### ISSUE TITLE
Replacement of TRANSLATE and CONDENSE does not work with macro placeholders

### ISSUE THREAD

**Issue by phenschke** [SAP/abap-cleaner#298]
Hello,

I encountered the following issue: when the cleaner encounters a TRANSLATE or CONDENSE statement that uses macro placeholders (e.g., "&1") as arguments, it raises an exception "_java.lang.IllegalArgumentException: Supplied TokenType IDENTIFIER unexpected for Token text '&1'; using inferred TokenType OTHER_OP!_" (at com.sap.adt.abapcleaner.parser.Token.createForAbap(Token.java:291)). 

As the error says, the issue is that "&1" is inferred to be of TokenType OTHER_OP, and not identifier in [Token:inferTypeFromAbapToken](https://github.com/SAP/abap-cleaner/blob/6d010e449895ed01c9d894c9e29d0e3724699e53/com.sap.adt.abapcleaner/src/com/sap/adt/abapcleaner/parser/Token.java#L408), which then leads to the exception in [Token:createForAbap](https://github.com/SAP/abap-cleaner/blob/6d010e449895ed01c9d894c9e29d0e3724699e53/com.sap.adt.abapcleaner/src/com/sap/adt/abapcleaner/parser/Token.java#L291).

Example:
```
DEFINE test_rule_macro.
    TRANSLATE &1 TO UPPER CASE.          " -> error!
    TRANSLATE &1 USING 'aabbaABBA'.      " -> error!
    CONDENSE &1.                         " -> error!

    MOVE &1 to &2.    " -> works! (if abap release >= 7.54)
    ADD &1 TO &2.     " -> works! (if abap release >= 7.54)

END-OF-DEFINITION.
```

On ABAP versions older than 7.54, this error will also occur for MOVE and ADD (and possibly lots of other rules that I didn't try out), because then [this else branch](https://github.com/SAP/abap-cleaner/blob/6d010e449895ed01c9d894c9e29d0e3724699e53/com.sap.adt.abapcleaner/src/com/sap/adt/abapcleaner/rules/commands/AddToEtcRule.java#L155C5-L155C10) will be entered.

In summary, the issue will arise for any call to Token.createForAbap() where the provided text starts with "&" and the provided type is IDENTIFIER. I guess a (maybe not so pretty) solution would be coding an exception for texts starting with "&" in Token:createForAbap() [I could provide a pull request for that].

Best regards,
Pascal

**Comment by jmgrassau** [comment]
Hi Pascal, 

nice one, thanks for reporting this and for your analysis! This will be fixed with the next release:

![image](https://github.com/SAP/abap-cleaner/assets/79639298/40550f75-c2f5-4301-ac2e-0009d55fb91b)

Kind regards,
Jörg-Michael

**Comment by jmgrassau** [comment]
Hi Pascal, 

thanks again for reporting this issue – this should now be fixed with version 1.16.1, which was just released!

Kind regards,
Jörg-Michael
